dashboard/app/models/levels/javalab.rb

```diff
@@ -47,6 +48,25 @@ def self.create_from_level_builder(params, level_params)
     )
   end
 
+  def parse_maze
+    if serialized_maze.nil? && csa_view_mode == 'neighborhood'
+      raise ArgumentError.new('neighborhood must have a serialized_maze')
+    end
+    return if serialized_maze.nil?
+    # convert maze into json object and validate each cell has a tileType
+    maze_json = serialized_maze.is_a?(Array) ? serialized_maze.to_json : serialized_maze
```

Comment:
Is there any reason (or could there be in the future) the serialized_maze wouldn't be of type Array or already in JSON?

Reply:
Looking at how it's defined for other levels, it's generally a string which we need to call `JSON.parse` on. I copied this behavior from how maze parses `serialized_maze`. I think the only way it would ever be defined as an array is if an engineeer was editing a level manually. Both are parseable (array or escaped string), so seems reasonable to allow both unless we have strong opinions otherwise.

Reply:
Gotcha- thanks for the context. That makes more sense. I was wondering if there would be any other formats to worry about, but it sounds like it's a pattern that's been working for maze already!